karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/replace.feature

```diff
@@ -63,4 +66,158 @@ Feature: Patch patient - Replace data
+      * def relationshipDetails = contactDetails[0].relationship
+      * def contactId = contactDetails[0].id
+      * def indexToUpdate = response.contact.findIndex(x => x.id == contactId)
+      * def contactPathToUpdateAndRemove = "/contact/" + indexToUpdate
+
+      # update emergency contact
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * def newMobileNumber = faker.phoneNumber()
+      * path 'Patient', nhsNumber
+      * request 
+      """
+      {
+        "patches":[
+          {"op":"replace",
+          "path":"#(contactPathToUpdateAndRemove)",
+          "value":{
+            "id": "#(contactId)",
+            "relationship":"#(relationshipDetails)",
+            "telecom":[
+              {"system":"phone",
+              "value":"#(newMobileNumber)"}
+              ]}}]}
+       """       
+      * method patch
+      * status 200
+      * def updatedContact = response.contact.find(x => x.id == contactId)
+      * def telecomValue = updatedContact.telecom[0].value
+      * def versionIdAftUpdate = response.meta.versionId
+      * match parseInt(versionIdAftUpdate) == parseInt(versionIdAftAdd) + 1
+      * match telecomValue == newMobileNumber
+     
+      
+      # remove emergency contact details
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * path 'Patient', nhsNumber
+    
+      * def patchRequest = 
+      """
+        {
+          "patches": [
+            {
+              "op": "test",
+              "path": "#(contactPathToUpdateAndRemove)",
+              "value":"#(updatedContact)"       
+            },
+            {
+            "op": "remove",
+              "path": "#(contactPathToUpdateAndRemove)"
+            }
+          ]
+        }
+      """
+      * request patchRequest
+      * method patch
+      * status 200
+      * def versionIdAftRemove = response.meta.versionId
+      * match parseInt(versionIdAftRemove) == parseInt(versionIdAftUpdate) + 1
+    Scenario: Healthcare worker can update communication language-interpreter details
+      * def nhsNumber = '5900071413'
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * path 'Patient', nhsNumber
+      * method get
+      * status 200
+      * def originalVersion = parseInt(response.meta.versionId)
+      * def commExtensionUrl = "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-NHSCommunication"
+      * def commLanguageDls = response.extension.find(x => x.url == commExtensionUrl)
+      * def interpreterDls = commLanguageDls[1]
+      * def commLanguageindex = response.extension.findIndex(x => x.url == commExtensionUrl)
+      * def interpreterPath = "/extension/" + commLanguageindex + "/extension/1"
```

Comment:
Again, seems like this test assumes a communication extension is already on 5900071413. Do you mind adding a comment explaining this as well please?

Reply:
Same like previous test, added a step to terminate the test if no communication extension on the response.

Reply:
The test should fail anyway if they patient isn't in the correct state, so I think a comment explaining when it's a problem with the data rather then the test is probably fine.
If you want to go down the route of failing the test, I think karate.abort would just skip it, and wouldn't show up in a test report. https://github.com/karatelabs/karate?tab=readme-ov-file#abort-and-fail